webapp/channels/src/components/post_view/channel_intro_message/channel_intro_message.tsx

```diff
@@ -38,19 +40,32 @@ type Props = {
     channelProfiles: UserProfileType[];
     enableUserCreation?: boolean;
     isReadOnly?: boolean;
+    isFavorite: boolean;
     teamIsGroupConstrained?: boolean;
     creatorName: string;
     teammate?: UserProfileType;
     teammateName?: string;
+    currentUser: UserProfileType;
     stats: any;
     usersLimit: number;
     channelMember?: ChannelMembership;
+    isMobileView: boolean;
     actions: {
         getTotalUsersStats: () => any;
+        favoriteChannel: (channelId: string) => any;
+        unfavoriteChannel: (channelId: string) => any;
     };
 }
 
 export default class ChannelIntroMessage extends React.PureComponent<Props> {
+    toggleFavorite = () => {
```

Comment:
Once this PR goes in, I'm going to make a ticket to make a reusable `toggleFavorite` action since I've definitely seen this pattern used in a few other places. It'd make a good HW ticket